twilight-gateway/src/shard.rs

```diff
@@ -1162,7 +647,11 @@ impl<Q: Queue> Shard<Q> {
             }
             Some(OpCode::Heartbeat) => {
                 tracing::debug!("received heartbeat");
-                self.next_action = Some(NextAction::Heartbeat);
+                self.pending = Pending::text(
+                    json::to_string(&Heartbeat::new(self.session().map(Session::sequence)))
+                        .unwrap(),
```

Comment:
If I remember correctly previous infallible `to_string` calls used `.expect`, I'd like at least some form of coherence